nextcord/application_command.py

```diff
@@ -1236,6 +1248,46 @@ def error_name(self) -> str:
         raise NotImplementedError
 
 
+class SlashOptionData:
+    """A class that contains data for an application command option.
+
+    These classes are only created for options that a user defined.
+    Thus, application command options that are not required, and weren't
+    given a value by a user will not have a dedicated instance of this class.
+
+    This class is for data recieving only! To create slash options, use :class:`SlashOption`.
+
+    Parameters
+    ----------
+    data: :class: Dict[Union[`str`, `int`, `bool`]]
+        The raw data for an option. See https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-application-command-interaction-data-option-structure for more information.
+    value: Union[`str`, `int`, `float`, `bool`]
+        Returns the user input value.
+    type: :class:`int`
+        The type of the application command option. This indicates what value the option accepts as an input
+    name: :class:`str`
+        The name of this option
+    focused: :class:`bool`
+        Whether the user is currently creating an input for this option. This is useful for :meth:`on_autocomplete`
+    """
+
+    __slots__: Tuple[str, ...] = ("data", "value", "type", "name", "focused")
+
+    def __init__(self, data) -> None:
```

Comment:
This type doesn't exist - Are you suggesting I should make a new type for this?

Reply:
https://github.com/nextcord/nextcord/blob/5d2eff1810edd76359b66f37f731537a520c3d2c/nextcord/types/interactions.py#L128-L135

It does exist, and it's already imported in this module.

Reply:
Must have missed that somehow. Either way, as part of that list, is the class  `_ApplicationCommandInteractionDataOptionSubcommand`, which the SlashOptionData class doesn't use. (It may possibly not use Snowflake either but I can't say this for sure). Would it be worth making a new dedicated type for this? or is there some typehinting approach I can take, to reassure pyright, that this class does, indeed, not get used?

Reply:
Well, it does get used to tell Pyright what keys and types of values a specific dictionary has. You could make a new union without the subcommand one, but I'm not sure what to call it.